src/Razor/src/Microsoft.VisualStudio.LanguageServices.Razor/LanguageClient/Cohost/CohostSignatureHelpEndpoint.cs

```diff
@@ -0,0 +1,98 @@
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Microsoft.VisualStudio.Razor.LanguageClient;
+using Microsoft.VisualStudio.Razor.LanguageClient.Cohost;
+using Microsoft.VisualStudio.Razor.LanguageClient.Extensions;
+
+namespace Microsoft.VisualStudio.LanguageServices.Razor.LanguageClient.Cohost;
+
+#pragma warning disable RS0030 // Do not use banned APIs
+[Shared]
+[CohostEndpoint(Methods.TextDocumentSignatureHelpName)]
+[Export(typeof(IDynamicRegistrationProvider))]
+[ExportCohostStatelessLspService(typeof(CohostSignatureHelpEndpoint))]
+[method: ImportingConstructor]
+#pragma warning restore RS0030 // Do not use banned APIs
+internal class CohostSignatureHelpEndpoint(
+    IRemoteServiceProvider remoteServiceProvider,
+    IHtmlDocumentSynchronizer htmlDocumentSynchronizer,
+    LSPRequestInvoker requestInvoker,
+    ILoggerFactory loggerFactory)
+    : AbstractRazorCohostDocumentRequestHandler<SignatureHelpParams, SignatureHelp?>, IDynamicRegistrationProvider
+{
+    private readonly IRemoteServiceProvider _remoteServiceProvider = remoteServiceProvider;
+    private readonly IHtmlDocumentSynchronizer _htmlDocumentSynchronizer = htmlDocumentSynchronizer;
+    private readonly LSPRequestInvoker _requestInvoker = requestInvoker;
+    private readonly ILogger _logger = loggerFactory.GetOrCreateLogger<CohostFoldingRangeEndpoint>();
+
+    protected override bool MutatesSolutionState => false;
+
+    protected override bool RequiresLSPSolution => true;
+
+    public Registration? GetRegistration(VSInternalClientCapabilities clientCapabilities, DocumentFilter[] filter, RazorCohostRequestContext requestContext)
+    {
+        if (clientCapabilities.TextDocument?.SignatureHelp?.DynamicRegistration == true)
+        {
+            return new Registration()
+            {
+                Method = Methods.TextDocumentSignatureHelpName,
+                RegisterOptions = new SignatureHelpRegistrationOptions()
+                {
+                    DocumentSelector = filter
+                }.EnableSignatureHelp()
+            };
+        }
+
+        return null;
+    }
+
+    protected override RazorTextDocumentIdentifier? GetRazorTextDocumentIdentifier(SignatureHelpParams request)
+        => request.TextDocument.ToRazorTextDocumentIdentifier();
+
+    protected async override Task<SignatureHelp?> HandleRequestAsync(SignatureHelpParams request, RazorCohostRequestContext context, CancellationToken cancellationToken)
+    {
+        var razorDocument = context.TextDocument.AssumeNotNull();
+
+        var data = await _remoteServiceProvider.TryInvokeAsync<IRemoteSignatureHelpService, RemoteSignatureHelp?>(
+            razorDocument.Project.Solution,
+            (service, solutionInfo, cancellationToken) => service.GetSignatureHelpAsync(solutionInfo, razorDocument.Id, request.Position.ToLinePosition(), request.Context.TriggerKind, request.Context.TriggerCharacter, cancellationToken),
+            cancellationToken).ConfigureAwait(false);
```

Comment:
>).ConfigureAwait(false) [](http://example.com/codeflow?start=29&length=23)

another super-nit :smile: - would've been more readable to me if closing paren with ConfigureAwait was on a the next line

Reply:
hmm.. closing paren on a line by itself? I would have thought the separate line would start with the `.`?

Reply:
I wouldn't have an issue with either. Vscode style enforcer wants closing paren on its own line - 

![image](https://github.com/dotnet/razor/assets/1863656/d7618c08-a8ee-40a6-a513-008ed1010750)

I just think that ConfigureAwait on the same line looks a bit funky, but even that is a super-nit, totally feel free to punt.

Reply:
Could we please not try to make our C# style somehow match VS Code's TypeScript style? We don't do that anywhere in the Razor code base, and I strongly feel that we should aim for idiomatically formatted C# code, and the prime example of that (IMO) is the Roslyn IDE code base.

If I were formatting the code highlighted by this comment, I would probably do something like so:

``` C#
var data = await _remoteServiceProvider
    .TryInvokeAsync<IRemoteSignatureHelpService, RemoteSignatureHelp?>(
            razorDocument.Project.Solution,
            (service, solutionInfo, cancellationToken) =>
                service.GetSignatureHelpAsync(solutionInfo, razorDocument.Id, request.Position.ToLinePosition(), request.Context.TriggerKind, request.Context.TriggerCharacter, cancellationToken),
            cancellationToken)
    .ConfigureAwait(false);
```

Placing a closing parenthesis on a separate line is not generally the way we've formatting argument lists in this code base, the Roslyn IDE code base, or even the .NET runtime code base.